### ISSUE TITLE
Target position goes forward, but stops at some time point.

### ISSUE THREAD

**Issue by victor0777** [esmini/esmini#523]
Hi, 

Thanks for your support.

Regarding a test-driver example code, I use it for testing my scenario.

[7.5. The ghost concept](https://esmini.github.io/#_the_ghost_concept)

I found that the target position(look_ahead position) goes forward well at first, but it stops at some time point. 

because the target position stops, and the speed is positive. steering value changes dramatically. so it turns.
[test-driver.zip](https://github.com/esmini/esmini/files/13997711/test-driver.zip)

Can you run my attached code and review it?

I don't understand why the target position is stuck.  

I traced global_pos_x, global_pos.y of SE_RoadInfo structure. They change well for a while, but they are not changing anymore.
typedef struct
{
    float global_pos_x;   // target position, in global coordinate system
    float global_pos_y;   // target position, in global coordinate system
    float global_pos_z;   // target position, in global coordinate system
...
} SE_RoadInfo;

**Comment by eknabevcc** [comment]
Thanks for reporting this finding. 

Root cause is that the car controlled by the application has a route assigned while initial/current position is far away from the route. So the function that maps the XY(world) positions from the controller to Road/Route positions fails since the position is not close to the defined route, instead it falls back to stay on current road (id 8). 

But I agree that is bad behavior. Commit 906bc1083ef0e952f549e42856d0069da5d32a8a change the strategy so that when current position is off route, the complete road network will be considered.

Actually while looking into this issue we found another bug in the XY2Road/Route mapping function now fixed by commit 66c53f6017f931293759a128253c152f06cf3c06.

Both commits are merged to dev and will be included in next release..

So, with that change your scenario and application will work as expected. I also made a simplified version of the scenario where the car will reach the route after some driving, and then also continue after reaching end of route.

[CarlaTown10HD_fixedTest_mod.zip](https://github.com/esmini/esmini/files/14122101/CarlaTown10HD_fixedTest_mod.zip)
